test/e2e/scheduling/ubernetes_lite.go

```diff
@@ -130,7 +130,7 @@ func SpreadServiceOrFail(f *framework.Framework, replicaCount int, image string)
 	framework.ExpectNoError(err)
 
 	// Now make sure they're spread across zones
-	zoneNames, err := e2enode.GetClusterZones(f.ClientSet)
+	zoneNames, err := e2enode.GetSchedulableClusterZones(f.ClientSet)
```

Comment:
IIUC, before https://github.com/kubernetes/enhancements/issues/3094, `PodTopologySpread` plugin will ignore taints in calculation, which means master nodes are also taken into account. So `GetClusterZones` helps to validate this logic.

Reply:
Is it going to allow Pods to be allocated to control plane nodes?

Reply:
Scheduling decisions are made by the Filter/Score results, not the role of the nodes. If control plane nodes have the untolerated taint like `node-role.kubernetes.io/master:NoSchedule`, they're still unschedulable. I have no idea whether the work nodes also have untolerated taints etc. but filtered out by `GetSchedulableClusterZones` which escaped from our validation in `checkZoneSpreading`.
https://github.com/kubernetes/kubernetes/blob/226323178e23b4c476001266beab1e2f116b3879/test/e2e/framework/node/resource.go#L600

Reply:
control plane nodes have untolerated taint and pods won't be allocated there. `GetClusterZones` returns all zones including ones control plane nodes are on. As a result, these control plane zones get no Pod and validation fails. Is my understanding correct?

Reply:
Ah, I misunderstood the intension of the tests. Narrow down the available zones make sense to me. Sorry for that.